crates/bins/src/bin/datadog_static_analyzer_server/ide/configuration_file/endpoints.rs

```diff
@@ -0,0 +1,63 @@
+use super::error::ConfigFileError;
+use super::models::{AddRuleSetsRequest, IgnoreRuleRequest};
+use super::static_analysis_config_file::StaticAnalysisConfigFile;
+use kernel::utils::encode_base64_string;
+use rocket::http::Status;
+use rocket::serde::json::Json;
+use tracing::instrument;
+
+#[instrument()]
+#[rocket::post(
+    "/v1/config/ignore-rule",
+    format = "application/json",
+    data = "<request>"
+)]
+pub fn ignore_rule(
+    request: Json<IgnoreRuleRequest>,
+) -> Result<Json<String>, rocket::response::status::Custom<ConfigFileError>> {
+    let IgnoreRuleRequest {
+        rule,
+        configuration_base64,
+        encoded,
+        ..
+    } = request.into_inner();
+    tracing::debug!(rule, content = &configuration_base64);
+    let result = StaticAnalysisConfigFile::with_ignored_rule(rule.into(), configuration_base64);
+    to_json_result(result, encoded)
+}
+
+#[instrument()]
+#[rocket::post("/v1/config/rulesets", format = "application/json", data = "<request>")]
+pub fn post_rulesets(
```

Comment:
There are cases where we shouldn't offer to onboard the user to a new languge, eg: if the config file uses a version greater than the current known version, if there are [exclusions applied](https://docs.google.com/document/d/1jjloCvTOAvK99WDbfdcnttjYtjq2MJfPE8YKvDv432s/edit#heading=h.ndmldug56n2w) or if the file is invalid. 

We might need a new `can_onboard` endpoint for that.

Reply:
We might, I agree. Although we could also consider other options like expanding the `get_rulesets` returning collection with that information.

Reply:
Note: After discussing about this in one of our RFCs, @albertvaka pointed out that it would be better to avoid conflating rulesets and the fact of being able to onboard, because this can change in the future.